libagent/device/ledger.py

```diff
@@ -81,7 +81,7 @@ def sign(self, identity, blob):
         if identity.curve_name == 'nist256p1':
             p2 = '81' if identity.identity_dict['proto'] == 'ssh' else '01'
         else:
-            p2 = '82' if identity.identity_dict['proto'] == 'ssh' else '02'
+            p2 = '82' if (identity.identity_dict['proto'] == 'ssh' and len(blob) != 32) else '02'
```

Comment:
this line is also somewhat mysterious to me

Reply:
So I guess that again if the blob is 32 bytes, even if the proto says to do ssh, we select the p2 bytes corresponding to a non-ssh interaction